examples/start_stop_server/start_stop_server.cpp

```diff
@@ -0,0 +1,36 @@
+//
+// Example how to use C API of mavsdk_server.
+//
+
+#include <mavsdk/mavsdk.h>
+#include <mavsdk/mavsdk_server/mavsdk_server_api.h>
+#include <iostream>
+#include <csignal>
+
+using namespace mavsdk;
+
+static struct MavsdkServer* mavsdk_server;
+
+void signal_handler(int sig)
+{
+    std::cout << "Received SIGUSR1 signal" << std::endl;
+
+    mavsdk_server_stop(mavsdk_server);
+}
+
+int main(int argc, char* argv[])
+{
+    if (argc != 2) {
+        std::cout << "Connection string required as argument" << std::endl;
+        return 1;
+    }
+
+    signal(SIGUSR1, signal_handler);
+
+    mavsdk_server_init(&mavsdk_server);
+    mavsdk_server_run(mavsdk_server, argv[1], 50051);
+
+    mavsdk_server_destroy(mavsdk_server);
```

Comment:
Shouldn't you `mavsdk_server_stop` before?

Reply:
`mavsdk_server_stop` is done in the signal handler.

Reply:
But then shouldn't you call `mavsdk_server_attach()` before? Maybe I'm misremembering it, but here it feels like it will destroy as soon as the autopilot is detected and `mavsdk_server_run` returns 🤔